Sources/ZIPFoundation/Archive.swift

```diff
@@ -152,6 +152,7 @@ public final class Archive: Sequence {
             self.endOfCentralDirectoryRecord = endOfCentralDirectoryRecord
         case .create:
             guard !fileManager.fileExists(atPath: url.path) else { return nil }
+            do { try fileManager.createParentDirectoryStructure(for: url) } catch { return nil }
```

Comment:
If I understand this correctly, this was added to implicitly create intermediate directories. I'd prefer the previous behaviour, where archive creation fails if intermediate directories are not present.  
While this might be less convenient in some cases, it avoids unintentional folder creation and it informs the API user about potential problems with the provided path.  

The existing behaviour is also more in-line with Foundation APIs like `FileManager` where missing intermediate directories also result in creation errors: 
```swift
let fm = FileManager()
let tempURL = URL(fileURLWithPath: NSTemporaryDirectory())
let didCreate = fm.createFile(atPath: tempURL.appendingPathComponent(UUID().uuidString).path, contents: nil, attributes: nil)
let nestedTempURL = tempURL.appendingPathComponent("test")
let didCreateNested = fm.createFile(atPath: nestedTempURL.appendingPathComponent(UUID().uuidString).path, contents: nil, attributes: nil)
print("didCreate:\(didCreate) didCreateNested:\(didCreateNested)")
```